cyvlfeat/fisher/cyfisher.pyx

```diff
@@ -59,17 +83,24 @@ cpdef cy_fisher(float[:, :] X,
             n_data, n_clusters, n_dimensions, 2 * n_clusters * n_dimensions,
             square_root, normalized, fast))
 
-    cdef float[:] enc = np.zeros(2 * n_clusters * n_dimensions,
-                                 dtype=np.float32)
+    enc = np.zeros(2 * n_clusters * n_dimensions,
+                                 dtype=X.dtype)
+
+
+    data = <void*> X.data
```

Comment:
I'm not keen on this syntax. Also, given [the data attribute](http://docs.scipy.org/doc/numpy-dev/release.html#assignment-of-ndarray-object-s-data-attribute) is deprecated (at least setting it), I don't think it is recommended to do this. Also, passing around `void*` is never friendly. According to the [Cython documentation](http://docs.cython.org/src/userguide/memoryviews.html#coercion-to-numpy), `np.asarray` does not copy the buffer and thus has little overhead. Also, given this [stackoverflow thread](http://stackoverflow.com/questions/18462785/what-is-the-recommended-way-of-allocating-memory-for-a-typed-memory-view) it seems like allocating memory is the least of our worries given the lack of tight loops in this code.